### DISCUSSION TITLE
Model serves probabilties instead of classification

### DISCUSSION THREAD

**Discussion by LukasLohm** [mlflow/mlflow#6839] [Q&A] [answered]
Hello everyone,

I'm currently using Mlflow with a xgboost model and want to predict probabilities. Even though I'm using the predict_proba method my model delivers still a classification:

`with mlflow.start_run():
    
    scale_pos_weight = 5
    
    xgbc_5 = XGBClassifier(random_state=0,scale_pos_weight=scale_pos_weight)

    xgbc_5.fit(X_train, y_train)
    y_pred = xgbc_5.predict_proba(X_test)
    
    #Accuracy_Score = accuracy_score(y_test, y_pred)
    #Recall_Score = recall_score(y_test, y_pred)
    #roc_auc_Score = roc_auc_score(y_test, y_pred)
    #precision_Score = precision_score(y_test, y_pred)
    
    #mlflow.log_metric("Roc_Auc", roc_auc_Score)
    #mlflow.log_metric("Recall_Score", Recall_Score)
    #mlflow.log_metric("Accuracy_Score", Accuracy_Score)
    #mlflow.log_metric("Precision_Score", precision_Score)
    mlflow.log_param("Scale_Pos_Weight", scale_pos_weight)
   
    modelpath = "/Users/kcz@qlik.com/mlflow/Model6"
    mlflow.sklearn.log_model(xgbc_5, "model")
    mlflow.sklearn.save_model(xgbc_5, modelpath)

    #accuracy.append(round(accuracy_score(y_test, y_pred),4))
    #recall.append(round(recall_score(y_test, y_pred),4))
    #roc_auc.append(round(roc_auc_score(y_test, y_pred),4))
    #precision.append(round(precision_score(y_test, y_pred),4))

    model_names = ['XGBoost_adjusted_weight_5']
    #result_df6 = pd.DataFrame({'Accuracy':accuracy,'Recall':recall, 'Roc_Auc':roc_auc, 'Precision':precision}, index=model_names)
    #result_df6` 
    
 Is there anything I can do to get probabilities?

**Comment by JHibbard** [comment, marked as answer]
Hi Lukas,

This is now possible in [MLflow 1.29.0](https://github.com/mlflow/mlflow/pull/6631) using the `pyfunc_predict_fn` argument in log_model!

```Python
import mlflow  # must be 1.29.0 or greater
from sklearn.datasets import load_wine
from xgboost import XGBClassifier


wine = load_wine()
X = wine.data
y = wine.target
 
with mlflow.start_run():
  xgb = XGBClassifier(random_state=0, use_label_encoder=False)
  xgb.fit(X, y)

  # Note: specify your prediction function here
  mlflow.sklearn.log_model(xgb, 'model', pyfunc_predict_fn='predict_proba', )


# Note: replace with your model location
logged_model = '...72220be6946/model'

# Load model as a PyFuncModel.
loaded_model = mlflow.pyfunc.load_model(logged_model)

# Predict on a Pandas DataFrame.
import pandas as pd
loaded_model.predict(pd.DataFrame(X))
```
